ciphey/basemods/Checkers/quadgrams.py

```diff
@@ -0,0 +1,54 @@
+import logging
+import re
+from math import log10
+from typing import Dict, Optional
+
+from ciphey.iface import Checker, Config, ParamSpec, T, Translation, registry
+from rich.logging import RichHandler
+
+
+@registry.register
+class Quadgrams(Checker[str]):
+
+    """
+    Uses Quadgrams to determine plaintext
+    """
+
+    def check(self, ctext: T) -> Optional[str]:
+        logging.debug("Trying Quadgrams checker")
+        ctext = re.sub("[^A-Z]", "", ctext.upper())
+        quadgrams = self.QUADGRAMS_DICT
+        N = sum(quadgrams.values())
+        score = 0
+        for key in quadgrams.keys():
+            quadgrams[key] = float(quadgrams[key]) / N
+        floor = log10(0.01 / N)
+        for i in range(len(ctext) - 4 + 1):
+            if ctext[i : i + 4] in quadgrams:
+                score += quadgrams[ctext[i : i + 4]]
+            else:
+                score += floor
+        if len(ctext) > 0:
+            score = score / len(ctext)
+        logging.info(f"Quadgrams is {score}")
+        if score > 0.00011:
```

Comment:
Why this score? Can we parametrise this as a `--C` checker argument or?

Reply:
That's the score that I found to be the best after lots of testing. Also yes we can make it changeable but it would be under `-p` not `-C`